### PR TITLE
News nac

### PR DESCRIPTION
**Pull request by rickwolox**
### Summary
Adding navigation android component and share element transition

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt [hunk 1]
```diff
@@ -58,7 +58,7 @@ class NewsFragment @Inject constructor() : WolmoFragment<NewsPresenter>(), INews
         })
         vNewsRecycler.addOnItemClickListener(object : OnItemClickListener {
             override fun onItemClicked(position: Int, view: View) {
-                presenter.onSelectedItem(position)
+                presenter.onSelectedItem(position, (view.findViewById(R.id.vNewLikeIcon) as ImageView))
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt]: It's necessary to use findViewById and cast it to ImageView? Have you tried to use directly vNewLikeIcon? However, I don't think that passing a View to the presenter to it's a good idea. Maybe the "extras" var that you create on presenter could be done on the fragment.

File: app/src/main/java/ar/com/wolox/android/example/ui/newdetail/NewDetailFragment.kt [hunk 2]
```diff
@@ -20,6 +22,14 @@ class NewDetailFragment : WolmoFragment<NewDetailPresenter>(), INewDetailView {
 
     @Inject internal lateinit var toastFactory: ToastFactory
 
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/newdetail/NewDetailFragment.kt:27]: Try to leave layout as the first method on class

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt [hunk 3]
```diff
@@ -96,10 +96,8 @@ class NewsFragment @Inject constructor() : WolmoFragment<NewsPresenter>(), INews
         newsAdapter.setUserId(userId)
     }
 
-    override fun onItemNewClicked(new: New) {
-        val intent = Intent(requireActivity(), NewDetailActivity::class.java)
-        intent.putExtra(NEW, new)
-        startActivity(intent)
+    override fun onItemNewClicked(bundle: Bundle, extras: FragmentNavigator.Extras) {
+        Navigation.findNavController(view!!).navigate(NAVIGATION_TO_NEW_DETAIL, bundle, null, extras)
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt]: We are going to do use some Kotlin magic here. We are going to create a separate file called NavigationExtensions.kt. There we will define the following:

`fun Fragment.requireView() = view!!

fun Fragment.requireNavController() = Navigation.findNavController(requireView())`

So, you can avoid doing` Navigation.findNavController(view!!)` and replace it with `requireNavController()`

**Review by lmiotti — CHANGES_REQUESTED** [review]
It's almost good. Some things to be done before merging!

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt (hunk 1 above)
Comment (rickwolox) [reply app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt]: I pass the reference of the view, for sharedElement transition to be done with the recycler items

**Review by lmiotti — APPROVED** [review]
well done

**Pull request merged**